### PR TITLE
Migrate analytics to use numbered values.

### PR DESCRIPTION
**Pull request by milkypostman**
In order to save space on analytics requests we now use numbers instead
of strings.

Also, we now store the names of events as their enum name in all
caps instead of lowercase.

### CODE REVIEW

**Comment by milkypostman** [comment]
@jiayliu @chuckhays

File: src/app_engine/analytics.py [hunk 1]
```diff
@@ -16,11 +16,19 @@
...
   # Event signifying that a room enters the state of having exactly
   # two participants.
-  ROOM_SIZE_2 = 'room_size_2'
-  ICE_CONNECTION_STATE_CONNECTED = 'ice_connection_state_connected'
+  ROOM_SIZE_2 = 2
+  ICE_CONNECTION_STATE_CONNECTED = 3
+
+  # Reverse map from value to enum name. Used to determine the logged value.
+  Name = {
+      2: 'ROOM_SIZE_2',
```
Comment (jiayliu) [inline src/app_engine/analytics.py]: Can we do 
ROOM_SIZE_2: 'ROOM_SIZE_2'?

Comment (milkypostman) [inline src/app_engine/analytics.py]: oh yes, of course. haha

File: src/app_engine/analytics.py [hunk 2]
```diff
@@ -61,9 +69,22 @@ def _timestamp_from_millis(self, time_ms):
...
+      room_id: Room ID related to the given event type.
+      time_ms: Time that the event occurred on the server. Will be automatically
+               populated if not given explicitly.
+      client_time_ms: Time that an event occurred on the client, if the event
+                      originated on the client.
+      host: Hostname this is being logged on.
+    """
+    # Be forgiving. If an event is a string or is an unknown number we
+    # still log it but log it as the string value.
+    event_type = EventType.Name.get(event_type_id, str(event_type_id))
```
Comment (jiayliu) [inline src/app_engine/analytics.py]: nit: rename event_type to event_type_name?

File: src/app_engine/analytics.py (hunk 1 above)
Comment (jiayliu) [inline src/app_engine/analytics.py]: lgtm with nits

On Mon, Mar 9, 2015 at 1:33 PM, Donald Ephraim Curtis <
notifications@github.com> wrote:

> In src/app_engine/analytics.py
> <https://github.com/webrtc/apprtc/pull/111#discussion_r26072718>:
>
> >  class EventType(object):
> >    # Event signifying that a room enters the state of having exactly
> >    # two participants.
> > -  ROOM_SIZE_2 = 'room_size_2'
> > -  ICE_CONNECTION_STATE_CONNECTED = 'ice_connection_state_connected'
> > +  ROOM_SIZE_2 = 2
> > +  ICE_CONNECTION_STATE_CONNECTED = 3
> > +
> > +  # Reverse map from value to enum name. Used to determine the logged value.
> > +  Name = {
> > +      2: 'ROOM_SIZE_2',
>
> oh yes, of course. haha
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/webrtc/apprtc/pull/111/files#r26072718>.
>



-- 
Jiayang Liu

**Comment by milkypostman** [comment]
Done.

**Comment by milkypostman** [comment]
PTAL

**Pull request merged**